### PR TITLE
README folder cleanup

### PR DESCRIPTION
**Pull request by bernardkwok**
Markdown cleanup for source directory, libraries. Minor update for ShaderGeneration doc.

### CODE REVIEW

File: documents/DeveloperGuide/ShaderGeneration.md [hunk 1]
```diff
@@ -1,12 +1,16 @@
 # Shader Generation
 
 ## 1.1 Scope
-Shader generation features are implemented as a core part of MaterialX. Most
-features are contained in the [MaterialXGenShader](/source/MaterialXGenShader) shared library, but some features are part of the [MaterialXCore](/source/MaterialXCore) library as well.
+Shader generation features are implemented as a core part of MaterialX. Core
+feature support is contained in the [MaterialXGenShader](/source/MaterialXGenShader) shared library, with some features part of the [MaterialXCore](/source/MaterialXCore) library.
```
Comment (niklasharrysson) [inline documents/DeveloperGuide/ShaderGeneration.md]: Maybe we could remove the last part of this sentence now, after the comma. The fact that we did updates to MaterialXCore in order to support our shader generation is not that relevant for a user at this point.

File: documents/Libraries/README.md [hunk 2]
```diff
@@ -32,4 +36,17 @@ The following is the layout of the definitions and implementations provided as p
...
+- Basic GLSL `lightshader` node definitions and implementations are provided for the following light types:
+    - point, directional, spot
+- Code generation is not currently supported for:
+    - `noise` nodes for `genglsl`
+    - `ambientocclusion` node for: `genosl` and `genglsl`.
+    - `arrayappend` node for: `genosl` and `genglsl`.
+    - `curveadjust` node for: `genosl` and `genglsl`.
+    - `displacementshader` and `volumeshader` nodes and associated operations (`add`, `multiply`, `mix`) for: `genosl` and `genglsl`.
+    - `add`, `multiply`, and `mix surfaceshader` node operations for `genglsl`.
+    - `PBR` library support in the reference `osl` implementation.
```
Comment (niklasharrysson) [inline documents/Libraries/README.md]: Not sure we need this last point. It's not "currently unsupported" as in _will be added later_. The genosl and genglsl implementations of pbrlib is kind of the reference.

File: documents/Libraries/README.md [hunk 3]
```diff
@@ -32,4 +36,17 @@ The following is the layout of the definitions and implementations provided as p
...
 
-Note that there is no reference implementation for the PBR library.
+## Support Notes:
+- GLSL language support is for version 4.0 or higher.
+- "default" color management support includes OSL and GLSL implementations for the following non-LUT transforms:
+    - lin_rec709, gamma18, gamma22, gamm24, acescg, srgb_texture
+- Basic GLSL `lightshader` node definitions and implementations are provided for the following light types:
+    - point, directional, spot
+- Code generation is not currently supported for:
+    - `noise` nodes for `genglsl`
```
Comment (niklasharrysson) [inline documents/Libraries/README.md]: Nitpick: Maybe name the nodes explicitly, noise2d/3d, cellnoise2d/3d, fractal3d

File: documents/Libraries/README.md [hunk 4]
```diff
@@ -32,4 +36,17 @@ The following is the layout of the definitions and implementations provided as p
         - [pbrlib_genosl_impl.mtlx](pbrlib/genosl/pbrlib_genosl_impl.mtlx) : Mapping from definitions to implementations
         -   OSL implementation files.
 
-Note that there is no reference implementation for the PBR library.
+## Support Notes:
+- GLSL language support is for version 4.0 or higher.
+- "default" color management support includes OSL and GLSL implementations for the following non-LUT transforms:
+    - lin_rec709, gamma18, gamma22, gamm24, acescg, srgb_texture
```
Comment (feldstj) [inline documents/Libraries/README.md]: minor: gamm24 should be gamma24

**Pull request merged**